common/persistence/visibility/elasticsearch/query/converter_test.go

```diff
@@ -0,0 +1,146 @@
+	"process_id >= 1":               `{"bool":{"filter":{"range":{"process_id":{"from":1,"include_lower":true,"include_upper":true,"to":null}}}}}`,
+	"process_id != 1":               `{"bool":{"must_not":{"match_phrase":{"process_id":{"query":1}}}}}`,
+	"process_id = 0 and status= 1 and channel = 4": `{"bool":{"filter":[{"match_phrase":{"process_id":{"query":0}}},{"match_phrase":{"status":{"query":1}}},{"match_phrase":{"channel":{"query":4}}}]}}`,
+	"process_id > 1 and status = 1":                `{"bool":{"filter":[{"range":{"process_id":{"from":1,"include_lower":false,"include_upper":true,"to":null}}},{"match_phrase":{"status":{"query":1}}}]}}`,
+	"id > 1 or process_id = 0":                     `{"bool":{"should":[{"range":{"id":{"from":1,"include_lower":false,"include_upper":true,"to":null}}},{"match_phrase":{"process_id":{"query":0}}}]}}`,
+	"id > 1 and d = 1 or process_id = 0 and x = 2": `{"bool":{"should":[{"bool":{"filter":[{"range":{"id":{"from":1,"include_lower":false,"include_upper":true,"to":null}}},{"match_phrase":{"d":{"query":1}}}]}},{"bool":{"filter":[{"match_phrase":{"process_id":{"query":0}}},{"match_phrase":{"x":{"query":2}}}]}}]}}`,
+	"(id > 1 and d = 1)":                           `{"bool":{"filter":[{"range":{"id":{"from":1,"include_lower":false,"include_upper":true,"to":null}}},{"match_phrase":{"d":{"query":1}}}]}}`,
+	"(id > 1 and d = 1) or (c=1)":                  `{"bool":{"should":[{"bool":{"filter":[{"range":{"id":{"from":1,"include_lower":false,"include_upper":true,"to":null}}},{"match_phrase":{"d":{"query":1}}}]}},{"match_phrase":{"c":{"query":1}}}]}}`,
+	"id > 1 or (process_id = 0)":                   `{"bool":{"should":[{"range":{"id":{"from":1,"include_lower":false,"include_upper":true,"to":null}}},{"match_phrase":{"process_id":{"query":0}}}]}}`,
+	"id in (1,2,3,4)":                              `{"bool":{"filter":{"terms":{"id":[1,2,3,4]}}}}`,
+	"a = 'text'":                                   `{"bool":{"filter":{"match_phrase":{"a":{"query":"text"}}}}}`,
+	"a like '%a%'":                                 `{"bool":{"filter":{"match_phrase":{"a":{"query":"a"}}}}}`,
+	"`by` = 1":                                     `{"bool":{"filter":{"match_phrase":{"by":{"query":1}}}}}`,
+	"id not like '%aaa%'":                          `{"bool":{"must_not":{"match_phrase":{"id":{"query":"aaa"}}}}}`,
+	"id not in (1,2,3)":                            `{"bool":{"must_not":{"terms":{"id":[1,2,3]}}}}`,
+	"id is not null":                               `{"bool":{"filter":{"exists":{"field":"id"}}}}`,
+	"id is null":                                   `{"bool":{"must_not":{"exists":{"field":"id"}}}}`,
+	"id != missing":                                `{"bool":{"filter":{"exists":{"field":"id"}}}}`,
+	"id = missing":                                 `{"bool":{"must_not":{"exists":{"field":"id"}}}}`,
+	"value = '1'":                                  `{"bool":{"filter":{"match_phrase":{"value":{"query":true}}}}}`,
```

Comment:
'1' is treated as true, what about string value '1'?

Reply:
This is a bug. I can't differentiate these w/o knowledge of field type. `'true'` value has similar problem.